pkg/rbac/namespacedAuthorization.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Contributors to the Open Cluster Management project
+package rbac
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+type userData struct {
+	// impersonate kubernetes.Interface //client with impersonation config
+	err        error
+	namespaces []string
+	// resources  map[string][]string //key:namespace value list of resources
+	updatedAt time.Time
+	lock      sync.Mutex
+}
```

Comment:
- Remove namespaces because we don't have a way to get an accurate list of namespaces for the user.
- Note that some of the fields aren't implemented with this PR. Keep those commented out for now.
 
```go
// Contains data about the resources the user is allowed to access.
type userData struct {
	// clusters     []string              // Managed clusters where the user has view access.
	// csResources  []resource            // Cluster-scoped resources on hub the user has list access.
	// nsResources  map[string][]resource // Namespaced resources on hub the user has list access. 
                                              //   key:namespace value list of resources
	
	// Internal fields to manage the cache.
	clustersErr       error      // Error while updating clusters data.
	clustersLock      sync.Mutex // Locks when clusters data is being updated.
	clustersUpdatedAt time.Time  // Time clusters was last updated.
	csrErr             error      // Error while updating cluster-scoped resources data.
	csrLock            sync.Mutex // Locks when cluster-scoped resources data is being updated.
	csrUpdatedAt       time.Time  // Time cluster-scoped resources was last updated.
	nsrErr             error      // Error while updating namespaced resources data.
	nsrLock            sync.Mutex // Locks when namespaced resources data is being updated.
	nsrUpdatedAt       time.Time  // Time namespaced resources was last updated.

	// impersonate *kubernetes.Interface // client with impersonation config
}
```

Reply:
We already define some of these in the sharedResources cache. Do we need to define them in both caches?